haystack/utils/openai_utils.py

```diff
@@ -79,6 +101,10 @@ def _openai_text_completion_tokenization_details(model_name: str):
         max_tokens_limit = 4000
         if USE_TIKTOKEN:
             tokenizer_name = MODEL_TO_ENCODING.get(model_name, "p50k_base")
+    elif "gpt-3.5-turbo" in model_name:
+        max_tokens_limit = 4096
+        if USE_TIKTOKEN:
+            tokenizer_name = MODEL_TO_ENCODING.get(model_name, "cl100k_base")
```

Comment:
Hey as a heads up this [PR](https://github.com/deepset-ai/haystack/pull/4331/files) will be conflict with this one. @vblagoje I think it would be worth the merge [PR](https://github.com/deepset-ai/haystack/pull/4331/files) first before because it is a better way to load the correct tokenizer.